code/modules/mining/ore_satchel.dm

```diff
@@ -0,0 +1,71 @@
+/obj/item/storage/bag/ore
+	name = "mining satchel"
+	desc = "This little bugger can be used to store and transport ores."
+	description_info = "You can attach a warp extraction pack to it, then click on an ore box that has a warp extraction beacon signaller attached to it to link them. Then ore put into this will be bluespace teleported into the ore box."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "satchel"
+	slot_flags = SLOT_BELT | SLOT_POCKET
+	w_class = 3
+	max_storage_space = 100
+	max_w_class = 3
+	can_hold = list(/obj/item/ore)
+	var/obj/structure/ore_box/linked_box
+	var/linked_beacon = FALSE // can't hold an actual beacon beclause storage code a shit
+	var/linked_beacon_uses = 3 // to hold the amount of uses the beacon had, storage code a shit.
+
+/obj/item/storage/bag/ore/examine(mob/user)
+	..()
+	if(user.Adjacent(src))
+		to_chat(user, FONT_SMALL(SPAN_NOTICE("It has a <b>warp extraction pack</b> inside.")))
+
+/obj/item/storage/bag/ore/drone
+	// this used to be 400. The inventory system FUCKING DIED at this.
+	max_storage_space = 200
+
+/obj/item/storage/bag/ore/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/extraction_pack))
+		var/obj/item/extraction_pack/E = W
+		if(linked_beacon)
+			to_chat(user, SPAN_WARNING("\The [src] already has a warp extraction pack!"))
+			return
+		linked_beacon = TRUE
+		linked_beacon_uses = E.uses_left
+		to_chat(user, SPAN_NOTICE("You attach \the [E] to \the [src]."))
+		qdel(E)
+	else if(W.isscrewdriver())
+		if(!linked_beacon)
+			to_chat(user, SPAN_WARNING("\The [src] doesn't have a linked extraction pack!"))
+			return
+		var/obj/item/extraction_pack/E = new /obj/item/extraction_pack(get_turf(user))
+		E.uses_left = linked_beacon_uses
```

Comment:
if you're gonna do this, then why not just make a typecasted variable to put the extraction pack in?

Reply:
the original implementation had that, but when i tested it in-game, the forceMoved extraction pack appeared in the ore inventory, since storage inventory shows contents. this is why i had to use this method instead